### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by tanishk15sharma**
- [x] User profile Page
- [x] User posts on his/her profile page

### CODE REVIEW

File: src/components/Feed.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
 import React, { useState, useEffect } from "react";
 import { PostCard } from "./PostCard";
-import { getAllPosts } from "../utils/posts";
-const Feed = () => {
+import { getAllPosts, getUserAllPosts } from "../utils/posts";
+const Feed = ({ userId }) => {
   const [posts, setPosts] = useState([]);
-
+  console.log(userId, "feed page me");
```
Comment (Pratham82) [inline src/components/Feed.js]: Remove logs

**Review by Pratham82 — APPROVED** [review]

File: src/components/SideNav.js [hunk 2]
```diff
@@ -20,10 +23,12 @@ const SideNav = () => {
           <span className="material-icons mr-4">bookmark_border</span>
           <span>Bookmarks</span>
         </li>
+        {/* <Link to={`/profile/${post.userId}`}> */}
```
Comment (deepak29-git) [inline src/components/SideNav.js:26]: remove commented code

File: src/pages/Profile.js [hunk 3]
```diff
@@ -1,52 +1,70 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import { Header } from "../components/Header";
```
Comment (deepak29-git) [inline src/pages/Profile.js:2]: you can put all the import in one file so your code looks more cleaner

File: src/pages/Profile.js [hunk 4]
```diff
@@ -1,52 +1,70 @@
...
+  const paramsUserId = useParams().id;
 
+  useEffect(() => {
+    (async () => {
+      const newUser = await getUser(paramsUserId);
+      setUser(newUser);
+    })();
+  }, [paramsUserId]);
+
+  console.log(user);
```
Comment (deepak29-git) [inline src/pages/Profile.js:22]: remove console after debugging

**Review by deepak29-git — APPROVED** [review]

**Review by JantuDeb — APPROVED** [review]
This is looks fine, clean and structured code

File: src/components/Feed.js (hunk 1 above)
Comment (tanishk15sharma) [reply src/components/Feed.js]: yes I will , actually I was checking `Id`

File: src/components/SideNav.js (hunk 2 above)
Comment (tanishk15sharma) [reply src/components/SideNav.js:26]: yes

File: src/pages/Profile.js (hunk 3 above)
Comment (tanishk15sharma) [reply src/pages/Profile.js:2]: Yes Thank you, but actually I dont have more than 2-3 files .

**Pull request merged**